test/switch.civet

```diff
@@ -222,3 +238,112 @@ describe "switch", ->
         }
       }
     """
+
+  describe "pattern matching", ->
+    testCase """
+      basic
+      ---
+      switch x
+        1
+          console.log "it's one"
+        2
+          console.log "it's two"
+      ---
+      if(x === 1) {
+          console.log("it's one")}
+      else if(x === 2) {
+          console.log("it's two")}
+    """
+
+    testCase.skip """
+      multiple cases
+      ---
+      switch x
+        1, 2
+          console.log "it's one or two"
+      ---
+      if(x === 1 || x === 2) {
+        console.log("it's one or two");
+      }
+    """
+
+    testCase """
+      array pattern
+      ---
+      switch x
+        [a, b]
+          console.log a, b
+      ---
+      if(Array.isArray(x) && x.length === 2) {
+          const [a, b] = x;
+          console.log(a, b)}
+    """
+
+    testCase """
+      array rest matcher
+      ---
+      switch x
+        [a, b...]
+          console.log a
+          console.log b
+      ---
+      if(Array.isArray(x) && x.length >= 1) {
+          const [a, ...b] = x;
+          console.log(a)
+          console.log(b)}
+    """
+
+    testCase """
+      array rest non end position
+      ---
+      switch x
+        [...a, b]
+          console.log a
+          console.log b
+      ---
+      const splice: <T>(this: T[], start: number, deleteCount?: number) => T[] = [].splice as any
+      if(Array.isArray(x) && x.length >= 1) {
+          const [...a] = x, [b] = splice.call(a, -1);
+          console.log(a)
+          console.log(b)}
+    """
+
+    testCase """
+      array this binding
+      ---
+      switch x
+        [@a, @b]
+          console.log this
+      ---
+      if(Array.isArray(x) && x.length === 2) {
+          const [a, b] = x;
+          this.a = a;
+          this.b = b;
+          console.log(this)}
+    """
+
+    testCase """
+      array rest this binding
+      ---
+      switch x
+        [@a, @b...]
+          console.log this
+      ---
+      if(Array.isArray(x) && x.length >= 1) {
+          const [a, ...b] = x;
+          this.a = a;
+          this.b = b;
+          console.log(this)}
+    """
+
+    testCase.skip """
+      nested array pattern
+      ---
+      switch x
+        [a, [b, c]]
+          console.log a, b, c
+      ---
+      if(Array.isArray(x) && x.length === 2 && Array.isArray(x[1]) && x[1].length === 2) {
+          const [a, [b, c]] = x;
+          console.log(a, b, c)}
+    """
```

Comment:
Missing test cases for:
* `ObjectMatchingPattern`
* `RegularExpressionLiteral`

But maybe these only exists in the grammar and are not yet implemented.

Reply:
Yep, not yet implemented.